docs/docs/loading-data-asynchronously.md

```diff
@@ -0,0 +1,155 @@
+        {this.state.repos.map(repo =>
+          <div key={repo.id}>{repo.name}</div>
+        )}
+      </div>
+    );
+  }
+}
+```
+
+The component will perform an initial render without any of the network data. When the axios promise resolves, it calls `setState` and the component is rerendered.
+
+[Try it on CodePen.](http://codepen.io/dashtinejad/pen/wgzEXJ?editors=0011)
+
+## Updates
+
+If the props change, we might need to fetch new data for the updated props. The [`componentDidUpdate` lifecycle hook](/react/docs/react-component.html#componentdidupdate) is a good place to achieve this, since we may not need to fetch new data if the props that we're interested in have not changed.
+
+Building on the previous example, we will pass the username as a prop instead and fetch new repos when it changes:
+
+```javascript{7-11,,13-15,17-21,35,38,44-47,53}
+class Repos extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {repos: []};
+  }
+
+  fetchRepos() {
+    axios.get(`https://api.github.com/users/${this.props.username}/repos`)
+      .then(response => this.setState({ repos: response.data }))
+      .catch(error => console.log(error));
+  }
+
+  componentDidMount() {
+    this.fetchRepos();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.username != prevProps.username) {
+      this.fetchRepos();
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>Repos by {this.props.username}</h1>
+        {this.state.repos.map(repo =>
+          <div key={repo.id}>{repo.name}</div>
+        )}
+      </div>
+    );
+  }
+}
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {username: 'facebook'};
+  }
+
+  render() {
+    return (
+      <div>
+        <button onClick={() => this.setState({username: 'facebook'})}>Facebook</button>
+        <button onClick={() => this.setState({username: 'microsoft'})}>Microsoft</button>
+        <button onClick={() => this.setState({username: 'google'})}>Google</button>
+        <Repos username={this.state.username} />
+      </div>
+    );
+  }
+}
+
+ReactDOM.render(<App />, document.getElementById('app'))
+```
+
+[Try it on CodePen.](http://codepen.io/dashtinejad/pen/zNpzVW?editors=0011)
+
+## Cancellation
+
+An old promise can be pending when a newer promise fulfills. This can cause the old promise to override the result of the new one. If a promise is pending when a component is updated, the result of the first promise should be ignored before a new one is created.
+Some data fetching APIs allow you to cancel requests, and for axios, we use [Cancellation](https://github.com/mzabriskie/axios#cancellation) by token:
+
+```javascript{5-8,10-11,14-15,19-23}
+class Repos extends React.Component {
+  // removed for brevity
+
+  fetchRepos() {
+    // cancel the previous request
+    if (typeof this._source != typeof undefined) {
+      this._source.cancel('Operation canceled due to new request.')
```

Comment:
Is it possible that `this._source` has already completed by the time this line of code runs? That seems possible to me.